### PR TITLE
Add apply function to solution

### PR DESCRIPTION
**Pull request by yitzchak**
Add ability to have custom apply function in v2 message solutions.

Result of PR suggestion in #99

### CODE REVIEW

File: src/types.js [hunk 1]
```diff
@@ -21,10 +21,11 @@ export type Message = {
...
   reference: ?string,
   solutions?: Array<{
-    title: ?string,
-    position: Range,
-    currentText: ?string,
-    replaceWith: string,
+    title?: string,
+    position?: Range,
+    currentText?: string,
+    replaceWith?: string,
```
Comment (Arcanemagus) [inline src/types.js]: See https://github.com/steelbrain/linter/pull/1206/files#r71568991 for the problem with this.

File: src/intentions.js [hunk 2]
```diff
@@ -65,16 +65,20 @@ export default class Intentions {
 
   static applyFix(textBuffer: TextBuffer, version: 1 | 2, fix: Object) {
     const range = version === 1 ? fix.range : fix.position
-    const currentText = version === 1 ? fix.oldText : fix.currentText
-    const replaceWith = version === 1 ? fix.newText : fix.replaceWith
+    if (version === 1 || fix.replaceWith) {
```
Comment (Arcanemagus) [inline src/intentions.js]: Again, this entire logic assumes the callback is required.

**Comment by steelbrain** [comment]
Nice work @yitzchak 

Could you apply the same suggestion I mentioned here https://github.com/steelbrain/linter/pull/1206#issuecomment-234035035

**Comment by steelbrain** [comment]
LGTM Thanks!

**Pull request merged**